From 6ef7619dccd4eab1f78ba076d0036bcf63941b03 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Tue, 7 Nov 2023 15:29:06 +0100
Subject: [PATCH 0031/1695] plat-stm32mp1: activate
 CFG_CORE_HALT_CORES_ON_PANIC

Default halt the other core when panicking on STM32MP15x platforms.
SGI15 is used for this purpose.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 27295ca4fba4d235a8218a6fe2b6a8bd4ac0deb8 plat-stm32mp1: activate CFG_CORE_HALT_CORES_ON_PANIC]
Change-Id: Ifd10d7f4605063b3b35002be646b52e9e3d46588
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
[etienne: cherry-picked from 3b252c23784b84d6a3c6e457bd360bea863ee2df]
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
---
 core/arch/arm/plat-stm32mp1/conf.mk | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp1/conf.mk b/core/arch/arm/plat-stm32mp1/conf.mk
index 48db504de..bf1d59942 100644
--- a/core/arch/arm/plat-stm32mp1/conf.mk
+++ b/core/arch/arm/plat-stm32mp1/conf.mk
@@ -126,11 +126,13 @@ endif # CFG_STM32MP13
 ifeq ($(CFG_STM32MP15),y)
 $(call force,CFG_BOOT_SECONDARY_REQUEST,y)
 $(call force,CFG_DRIVERS_CLK_FIXED,n)
+$(call force,CFG_HALT_CORES_ON_PANIC_SGI,15)
 $(call force,CFG_SECONDARY_INIT_CNTFRQ,y)
 $(call force,CFG_STM32MP1_SHARED_RESOURCES,y)
 $(call force,CFG_STM32_SAES,n)
 $(call force,CFG_STM32MP15_CLK,y)
 CFG_CORE_RESERVED_SHM ?= n
+CFG_HALT_CORES_ON_PANIC ?= y
 CFG_EXTERNAL_DT ?= y
 CFG_STM32_BSEC_SIP ?= y
 CFG_TEE_CORE_NB_CORE ?= 2
-- 
2.47.2

